<a href="https://colab.research.google.com/github/Sangram-Rout/me314/blob/main/HW7_Problem_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np
import sympy as sym
from sympy.abc import t
%matplotlib inline
import matplotlib.pyplot as plt

#######################
# Custom latex printing
def custom_latex_printer(exp,**options):
    from google.colab.output._publish import javascript
    url = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.3/latest.js?config=TeX-AMS_HTML"
    javascript(url=url)
    return sym.printing.latex(exp,**options)
sym.init_printing(use_latex="mathjax",latex_printer=custom_latex_printer)

####################
# Simulation helpers
def integrate(f,x0,dt):
    """
    This function takes in an initial condition x0 and a timestep dt,
    as well as a dynamical system f(x) that outputs a vector of the
    same dimension as x0. It outputs a vector x at the future time step.
    """
    k1=dt*f(x0)
    k2=dt*f(x0+k1/2.)
    k3=dt*f(x0+k2/2.)
    k4=dt*f(x0+k3)
    xnew=x0+(1/6.)*(k1+2.*k2+2.*k3+k4)
    return xnew

def simulate(f,x0,tspan,dt):
    """
    This function takes in an initial condition x0, a timestep dt,
    a time span tspan consisting of a list [min_time, max_time],
    as well as a dynamical system f(x) that outputs a vector of the
    same dimension as x0. Additionally, this includes a flag (default false)
    that allows one to supply an Euler intergation scheme instead of 
    the given scheme. It outputs a full trajectory simulated
    over the time span of dimensions (xvec_size, time_vec_size).
    """
    N = int((max(tspan)-min(tspan))/dt)
    x = np.copy(x0)
    tvec = np.linspace(min(tspan),max(tspan),N)
    xtraj = np.zeros((len(x0),N))
    for i in range(N):
        xtraj[:,i]=integrate(f,x,dt)
        x = np.copy(xtraj[:,i])
    return xtraj   

In [ ]:
from sympy.matrices import MatrixSymbol, Transpose
from sympy.functions import transpose
from sympy.matrices import Matrix
from sympy import MatrixSymbol, Inverse

In [ ]:
# Here we define the constants/variables relevant to the problem

#Updated for double pendulum
m,R1,R2,R3,g,k1,k2, lamb, l, J, w, a, b= sym.symbols('m R1 R2 R3 g k1 k2 lambda l J w a b')

k1,k2,lbd1,lbd2= sym.symbols('k1 k2 lambda_1 lambda_2')
#m,m,
th1 = sym.Function(r'\theta_1')(t)
th2 = sym.Function(r'\theta_2')(t)
th3 = sym.Function(r'\theta_3')(t)
theta1 = sym.symbols(r'\theta_1')
theta1_dot = sym.symbols(r'\dot{\theta_1}')
theta1_ddot = sym.symbols(r'\ddot{\theta_1}')
theta2 = sym.symbols(r'\theta_2')
theta2_dot = sym.symbols(r'\dot{\theta_2}')
theta2_ddot = sym.symbols(r'\ddot{\theta_2}')
theta3 = sym.symbols(r'\theta_3')
theta3_dot = sym.symbols(r'\dot{\theta_3}')
theta3_ddot = sym.symbols(r'\ddot{\theta_3}')
a = sym.Function(r'a')(t)
b = sym.Function(r'b')(t)


# Here we define our configuration variables
#q = sym.Matrix([th1,th2,th3])
q = sym.Matrix([th1,th2])
qdot = q.diff(t)
qddot = qdot.diff(t)

l=1;
dt=0.01;
k1=10;
k2=20;
J=1;
m=1;
g=9.81;


# Coordinate change to make Lagrangian calculation easier
x1 = R1*sym.sin(th1)
y1 = -R1*sym.cos(th1)
x2 = x1 + R2*sym.sin(th1+th2)
y2 = y1 - R2*sym.cos(th1+th2)
#x3 = x2 + R3*sym.sin(th1+th2+th3)
#y3 = y2 - R3*sym.cos(th1+th2+th3)
x1d = x1.diff(t)
y1d = y1.diff(t)
x2d = x2.diff(t)
y2d = y2.diff(t)
#x3d = x3.diff(t)
#y3d = y3.diff(t)



In [ ]:
g12=sym.Matrix(4, 4, [1, 0, 0, a, 0, 1,0,b,0,0,1,0,0,0,0,1])
#display(g12)
#g1.dt=g1.diff(t)
#display(g1.dt)
#g1.inv=g1**(-1)
#display(g1.inv)

In [ ]:
#center of the body 1
g1=g12*g111
display(g1)

⎡cos(\theta₁(t))  -sin(\theta₁(t))  0  a(t)⎤
⎢                                          ⎥
⎢sin(\theta₁(t))  cos(\theta₁(t))   0  b(t)⎥
⎢                                          ⎥
⎢       0                0          1   0  ⎥
⎢                                          ⎥
⎣       0                0          0   1  ⎦

In [ ]:
g111=sym.Matrix(4, 4, [sym.cos(th1), -1*sym.sin(th1), 0, 0, 1*sym.sin(th1), sym.cos(th1),0,0,0,0,1,0,0,0,0,1])
#display(g111)
g111.dt=g1.diff(t)
#display(g1.dt)
g111.inv=g1**(-1)
#display(g1.inv)

In [ ]:
# V_b=g1.inv*g1.dt
# display(sym.simplify(V_b[2]))

In [ ]:
#position of the body 1 lower end
#g11=sym.Matrix(4, 4,[1 , 0,0,0, 0, 1,0, l*sym.cos(th1)/2+(l)/2,0, 0,1,0 ,0,0,0, 1])
#display(g11)



In [ ]:
#translation of the coordinate on the right side
#bring it to the center again
r32=sym.Matrix(4, 4,[1 , 0,0,0, 0, 1, 0, -l/2,0,0,1,0,0,0,0 , 1])
display(r32)

⎡1  0  0   0  ⎤
⎢             ⎥
⎢0  1  0  -0.5⎥
⎢             ⎥
⎢0  0  1   0  ⎥
⎢             ⎥
⎣0  0  0   1  ⎦

In [ ]:
############
###Coordinate on the right-side
r21=sym.Matrix(4, 4, [sym.cos(th2), -1*sym.sin(th2), 0,l/2*sym.sin(th2),  1*sym.sin(th2), sym.cos(-th2),0,-l/2*sym.cos(th2),0,0,1,0,0,0,0,1])
display(g1)
display(r32)
display(r21)
g22=g1*r32*r21
display(g22)


⎡cos(\theta₁(t))  -sin(\theta₁(t))  0  a(t)⎤
⎢                                          ⎥
⎢sin(\theta₁(t))  cos(\theta₁(t))   0  b(t)⎥
⎢                                          ⎥
⎢       0                0          1   0  ⎥
⎢                                          ⎥
⎣       0                0          0   1  ⎦

⎡1  0  0   0  ⎤
⎢             ⎥
⎢0  1  0  -0.5⎥
⎢             ⎥
⎢0  0  1   0  ⎥
⎢             ⎥
⎣0  0  0   1  ⎦

⎡cos(\theta₂(t))  -sin(\theta₂(t))  0  0.5⋅sin(\theta₂(t)) ⎤
⎢                                                          ⎥
⎢sin(\theta₂(t))  cos(\theta₂(t))   0  -0.5⋅cos(\theta₂(t))⎥
⎢                                                          ⎥
⎢       0                0          1           0          ⎥
⎢                                                          ⎥
⎣       0                0          0           1          ⎦

⎡-sin(\theta₁(t))⋅sin(\theta₂(t)) + cos(\theta₁(t))⋅cos(\theta₂(t))  -sin(\the
⎢                                                                             
⎢sin(\theta₁(t))⋅cos(\theta₂(t)) + sin(\theta₂(t))⋅cos(\theta₁(t))   -sin(\the
⎢                                                                             
⎢                                0                                            
⎢                                                                             
⎣                                0                                            

ta₁(t))⋅cos(\theta₂(t)) - sin(\theta₂(t))⋅cos(\theta₁(t))  0  a(t) + 0.5⋅sin(\
                                                                              
ta₁(t))⋅sin(\theta₂(t)) + cos(\theta₁(t))⋅cos(\theta₂(t))  0  b(t) + 0.5⋅sin(\
                                                                              
                       0                                   1                  
                                                   

In [ ]:
g_sc_dot = g22.diff(t)
g_sc_i = g22.inv()
v2_m = g_sc_i*g_sc_dot 
v2 = sym.simplify(sym.Matrix([v2_m[2,1],v2_m[0,2],v2_m[1,0],v2_m[0,3],v2_m[1,3],v2_m[2,3]]))

In [ ]:
display(v2)

⎡                                                                             
⎢                                                                             
⎢                                                                             
⎢                                                                             
⎢                                                                             
⎢                                                                             
⎢                                                                             
⎢                                                                             
⎢                                     d                                       
⎢0.5⋅sin(2⋅\theta₁(t) + 2⋅\theta₂(t))⋅──(b(t)) + 0.5⋅cos(\theta₁(t) + \theta₂(
⎢                                     dt                                      
⎢─────────────────────────────────────────────────────────────────────────────
⎢                                                   

In [ ]:
#for right hand side of the matrix

gls = sym.Matrix([[sym.cos(th1), -sym.sin(th1), 0, a],
                   [sym.sin(th1),  sym.cos(th1), 0, b],
                   [0,0,1,0],
                   [0,0,0,1]])
display(gls)
gls2 = sym.Matrix([[1, 0, 0, 0],
                   [0, 1, 0, -l/2],
                   [0, 0, 1, 0],
                   [0, 0, 0, 1]])
display(gls2)
gls3= sym.Matrix([[sym.cos(th3), -sym.sin(th3), 0, (l/2)*sym.sin(th3)],
                   [sym.sin(th3),  sym.cos(th3), 0, (-l/2)*sym.cos(th3)],
                   [0,0,1,0],
                   [0,0,0,1]])
display(gls3)
glsd= gls*gls2*gls3
display(glsd)

⎡cos(\theta₁(t))  -sin(\theta₁(t))  0  a(t)⎤
⎢                                          ⎥
⎢sin(\theta₁(t))  cos(\theta₁(t))   0  b(t)⎥
⎢                                          ⎥
⎢       0                0          1   0  ⎥
⎢                                          ⎥
⎣       0                0          0   1  ⎦

⎡1  0  0   0  ⎤
⎢             ⎥
⎢0  1  0  -0.5⎥
⎢             ⎥
⎢0  0  1   0  ⎥
⎢             ⎥
⎣0  0  0   1  ⎦

⎡cos(\theta₃(t))  -sin(\theta₃(t))  0  0.5⋅sin(\theta₃(t)) ⎤
⎢                                                          ⎥
⎢sin(\theta₃(t))  cos(\theta₃(t))   0  -0.5⋅cos(\theta₃(t))⎥
⎢                                                          ⎥
⎢       0                0          1           0          ⎥
⎢                                                          ⎥
⎣       0                0          0           1          ⎦

⎡-sin(\theta₁(t))⋅sin(\theta₃(t)) + cos(\theta₁(t))⋅cos(\theta₃(t))  -sin(\the
⎢                                                                             
⎢sin(\theta₁(t))⋅cos(\theta₃(t)) + sin(\theta₃(t))⋅cos(\theta₁(t))   -sin(\the
⎢                                                                             
⎢                                0                                            
⎢                                                                             
⎣                                0                                            

ta₁(t))⋅cos(\theta₃(t)) - sin(\theta₃(t))⋅cos(\theta₁(t))  0  a(t) + 0.5⋅sin(\
                                                                              
ta₁(t))⋅sin(\theta₃(t)) + cos(\theta₁(t))⋅cos(\theta₃(t))  0  b(t) + 0.5⋅sin(\
                                                                              
                       0                                   1                  
                                                   

In [ ]:
glsd_dot = glsd.diff(t)
glsd_inv = glsd.inv()
Vb3_hat = glsd_inv*glsd_dot 
Vb3 = sym.simplify(sym.Matrix([Vb3_hat[2,1],Vb3_hat[0,2],Vb3_hat[1,0],Vb3_hat[0,3],Vb3_hat[1,3],Vb3_hat[2,3]]))
display(Vb3)

⎡                                                                             
⎢                                                                             
⎢                                                                             
⎢                                                                             
⎢                                                                             
⎢                                                                             
⎢                                                                             
⎢                                                                             
⎢                                     d                                       
⎢0.5⋅sin(2⋅\theta₁(t) + 2⋅\theta₃(t))⋅──(b(t)) + 0.5⋅cos(\theta₁(t) + \theta₃(
⎢                                     dt                                      
⎢─────────────────────────────────────────────────────────────────────────────
⎢                                                   

In [ ]:
#display(sym.simplify(Vb2))

In [ ]:
#Vb2=sym.Matrix(6,1,[Vb2_hat[7],Vb2_hat[2],Vb2_hat[3],Vb2_hat[4],Vb2_hat[8],Vb2_hat[12]])


In [ ]:
I=  sym.Matrix(6,6,[1,0,0,0,0,0,   0,1,0,0,0,0 , 0,0,1,0,0,0   , 0,0,0,1,0,0,   0,0,0,0,1,0,  0,0,0,0,0,1])
display(I)

⎡1  0  0  0  0  0⎤
⎢                ⎥
⎢0  1  0  0  0  0⎥
⎢                ⎥
⎢0  0  1  0  0  0⎥
⎢                ⎥
⎢0  0  0  1  0  0⎥
⎢                ⎥
⎢0  0  0  0  1  0⎥
⎢                ⎥
⎣0  0  0  0  0  1⎦

In [ ]:
display(gls)

⎡cos(\theta₁(t))  -sin(\theta₁(t))  0  a(t)⎤
⎢                                          ⎥
⎢sin(\theta₁(t))  cos(\theta₁(t))   0  b(t)⎥
⎢                                          ⎥
⎢       0                0          1   0  ⎥
⎢                                          ⎥
⎣       0                0          0   1  ⎦

In [ ]:
gcenterdot = gls.diff(t)
gcenterinv = gls.inv()
va_centerM = gcenterinv*gcenterdot 
Vbcenter = sym.simplify(sym.Matrix([va_centerM[2,1],va_centerM[0,2],va_centerM[1,0],va_centerM[0,3],va_centerM[1,3],va_centerM[2,3]]))
display(Vbcenter)

⎡                          0                          ⎤
⎢                                                     ⎥
⎢                          0                          ⎥
⎢                                                     ⎥
⎢                   d                                 ⎥
⎢                   ──(\theta₁(t))                    ⎥
⎢                   dt                                ⎥
⎢                                                     ⎥
⎢                 d                          d        ⎥
⎢ sin(\theta₁(t))⋅──(b(t)) + cos(\theta₁(t))⋅──(a(t)) ⎥
⎢                 dt                         dt       ⎥
⎢                                                     ⎥
⎢                  d                          d       ⎥
⎢- sin(\theta₁(t))⋅──(a(t)) + cos(\theta₁(t))⋅──(b(t))⎥
⎢                  dt                         dt      ⎥
⎢                                                     ⎥
⎣                          0                          ⎦

In [ ]:
KE1=1/2*Vbcenter.T*I*Vbcenter
display(KE1)
Potential1=sym.Matrix([m*g*gls[7]])
display(Potential1)
Lag1=sym.simplify(KE1 - Potential1)

⎡                                                                             
⎢⎛                  d                          d       ⎞ ⎛                    
⎢⎜- sin(\theta₁(t))⋅──(a(t)) + cos(\theta₁(t))⋅──(b(t))⎟⋅⎜- 0.5⋅sin(\theta₁(t)
⎣⎝                  dt                         dt      ⎠ ⎝                    

                                                                              
  d                              d       ⎞   ⎛                    d           
)⋅──(a(t)) + 0.5⋅cos(\theta₁(t))⋅──(b(t))⎟ + ⎜0.5⋅sin(\theta₁(t))⋅──(b(t)) + 0
  dt                             dt      ⎠   ⎝                    dt          

                                                                              
                   d       ⎞ ⎛                d                          d    
.5⋅cos(\theta₁(t))⋅──(a(t))⎟⋅⎜sin(\theta₁(t))⋅──(b(t)) + cos(\theta₁(t))⋅──(a(
                   dt      ⎠ ⎝                dt                         dt   

                           2⎤
   ⎞       ⎛d      

[9.81⋅b(t)]

In [ ]:
KE2=1/2*v2.T*I*v2
display(sym.simplify(KE2))
Potential2=sym.Matrix([m*g*g22[7]])
display(Potential2)
Lag2=sym.simplify(KE2 - Potential2)

⎡⎛                                     2                                      
⎢⎜    ⎛d                d             ⎞    ⎛                                 d
⎢⎜0.5⋅⎜──(\theta₁(t)) + ──(\theta₂(t))⎟  + ⎜0.5⋅sin(\theta₁(t) + \theta₂(t))⋅─
⎢⎝    ⎝dt               dt            ⎠    ⎝                                 d
⎢─────────────────────────────────────────────────────────────────────────────
⎢                                                                             
⎣                                                                             

                                                                              
                               d                                              
─(a(t)) + 0.25⋅sin(\theta₂(t))⋅──(\theta₁(t)) - 0.5⋅cos(\theta₁(t) + \theta₂(t
t                              dt                                             
──────────────────────────────────────────────────────────────────────────────
                                                   

[9.81⋅b(t) + 4.905⋅sin(\theta₁(t))⋅sin(\theta₂(t)) - 4.905⋅cos(\theta₁(t))⋅cos
(\theta₂(t)) - 4.905⋅cos(\theta₁(t))]

In [ ]:
KE3=1/2*Vb3.T*I*Vb3
#display(sym.simplify(KE3))
Potential3=sym.Matrix([m*g*glsd[7]])
#display(Potential3)
Lag3=sym.simplify(KE3 - Potential3)

In [ ]:
Lag=Lag1+Lag2+Lag3


In [ ]:
phi1 = g22[1,3]-0.5*l*sym.cos(th2)
phi2 = glsd[1,3]-0.5*l*sym.cos(th3)
display(phi1)
display(phi2)

Fth2 = -k1*(th1-0)-k2*(th2-(sym.pi/20+(sym.pi/3)*sym.sin(t/2)*sym.sin(t/2)))
Fth3 = -k1*(th1-0)-k2*(th3-(-sym.pi/20-(sym.pi/3)*sym.sin(t/2)*sym.sin(t/2)))
display(Fth2)
display(Fth3)

b(t) + 0.5⋅sin(\theta₁(t))⋅sin(\theta₂(t)) - 0.5⋅cos(\theta₁(t))⋅cos(\theta₂(t
)) - 0.5⋅cos(\theta₁(t)) - 0.5⋅cos(\theta₂(t))

b(t) + 0.5⋅sin(\theta₁(t))⋅sin(\theta₃(t)) - 0.5⋅cos(\theta₁(t))⋅cos(\theta₃(t
)) - 0.5⋅cos(\theta₁(t)) - 0.5⋅cos(\theta₃(t))

                                         2⎛t⎞    
                                 20⋅π⋅sin ⎜─⎟    
                                          ⎝2⎠    
-10⋅\theta₁(t) - 20⋅\theta₂(t) + ──────────── + π
                                      3          

                                         2⎛t⎞    
                                 20⋅π⋅sin ⎜─⎟    
                                          ⎝2⎠    
-10⋅\theta₁(t) - 20⋅\theta₃(t) - ──────────── - π
                                      3          

In [ ]:
q = sym.Matrix([th2,th3])
qdot = q.diff(t)
qddot = qdot.diff(t)
display(q)

phi = sym.Matrix([lbd1*(phi1.diff(th2)),lbd2*(phi2.diff(th3))])
display(phi)

⎡\theta₂(t)⎤
⎢          ⎥
⎣\theta₃(t)⎦

⎡λ₁⋅(0.5⋅sin(\theta₁(t))⋅cos(\theta₂(t)) + 0.5⋅sin(\theta₂(t))⋅cos(\theta₁(t))
⎢                                                                             
⎣λ₂⋅(0.5⋅sin(\theta₁(t))⋅cos(\theta₃(t)) + 0.5⋅sin(\theta₃(t))⋅cos(\theta₁(t))

 + 0.5⋅sin(\theta₂(t)))⎤
                       ⎥
 + 0.5⋅sin(\theta₃(t)))⎦

In [ ]:

# Euler Lagrange Equations
dLdq = sym.Matrix([Lag]).jacobian(q).T
ddLdqdot_dt = sym.Matrix([Lag]).jacobian(qdot).T.diff(t)
EulerLagrangeEQs = sym.Eq(ddLdqdot_dt-dLdq,phi)


In [ ]:
dphi1dt = phi1.diff(t)
dphi2dt = phi2.diff(t)
dphi1ddt = dphi1dt.diff(t)
dphi2ddt = dphi2dt.diff(t)
Equation1 = sym.Eq(dphi1ddt,0)
Equation2 = sym.Eq(dphi2ddt,0)
ELsol = sym.solve([EulerLagrangeEQs,Equation1,Equation2],[qddot[0],qddot[1],lbd1,lbd2])


In [ ]:
display(ELsol)

⎧          ⎛                                                 ⎛                
⎪          ⎜                                                 ⎜                
⎪    0.005⋅⎜(sin(\theta₁(t) + \theta₂(t)) + sin(\theta₂(t)))⋅⎜12.5⋅(sin(\theta
⎪          ⎜                                                 ⎜                
⎨          ⎝                                                 ⎝                
⎪λ₁: ─────────────────────────────────────────────────────────────────────────
⎪                                                                             
⎪                                                                             
⎩                                                                             

                                                                              
                                                               ⎛d             
₁(t) + 3.0⋅\theta₂(t)) + sin(3.0⋅\theta₁(t) + 5.0⋅\theta₂(t)))⋅⎜──(\theta₁(t))
                                                   


$$\left \{ \lambda_{1} : \frac{0.005}{\left(\sin{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} + \sin{\left (\theta_{2}{\left (t \right )} \right )}\right)^{2} \cos^{3}{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )}} \left(\left(\sin{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} + \sin{\left (\theta_{2}{\left (t \right )} \right )}\right) \left(12.5 \left(\sin{\left (\theta_{1}{\left (t \right )} + 3.0 \theta_{2}{\left (t \right )} \right )} + \sin{\left (3.0 \theta_{1}{\left (t \right )} + 5.0 \theta_{2}{\left (t \right )} \right )}\right) \left(\frac{d}{d t} \theta_{1}{\left (t \right )}\right)^{2} + 50.0 \left(\sin{\left (3.0 \theta_{1}{\left (t \right )} + 3.0 \theta_{2}{\left (t \right )} \right )} + \sin{\left (5.0 \theta_{1}{\left (t \right )} + 5.0 \theta_{2}{\left (t \right )} \right )}\right) \left(\frac{d}{d t} a{\left (t \right )}\right)^{2} - 50.0 \left(\sin{\left (3.0 \theta_{1}{\left (t \right )} + 3.0 \theta_{2}{\left (t \right )} \right )} + \sin{\left (5.0 \theta_{1}{\left (t \right )} + 5.0 \theta_{2}{\left (t \right )} \right )}\right) \left(\frac{d}{d t} b{\left (t \right )}\right)^{2} - 50.0 \left(\cos{\left (3.0 \theta_{1}{\left (t \right )} + 3.0 \theta_{2}{\left (t \right )} \right )} - \cos{\left (5.0 \theta_{1}{\left (t \right )} + 5.0 \theta_{2}{\left (t \right )} \right )}\right) \frac{d}{d t} a{\left (t \right )} \frac{d}{d t} b{\left (t \right )} - 400.0 \sin^{2}{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos^{3}{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \frac{d}{d t} a{\left (t \right )} \frac{d}{d t} b{\left (t \right )} - 100.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \sin^{2}{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \left(\frac{d}{d t} b{\left (t \right )}\right)^{2} - 100.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \sin{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} b{\left (t \right )} - 100.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \sin{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} b{\left (t \right )} - 200.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \sin{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \frac{d}{d t} a{\left (t \right )} \frac{d}{d t} b{\left (t \right )} - 200.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \sin{\left (\theta_{2}{\left (t \right )} \right )} \cos^{3}{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} b{\left (t \right )} - 400.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos^{4}{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \left(\frac{d}{d t} a{\left (t \right )}\right)^{2} + 400.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos^{4}{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \left(\frac{d}{d t} b{\left (t \right )}\right)^{2} - 200.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos^{3}{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{2}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} a{\left (t \right )} - 25.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos^{2}{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{1}{\left (t \right )}\right)^{2} - 100.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \cos{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} a{\left (t \right )} - 50.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{1}{\left (t \right )}\right)^{2} - 100.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} a{\left (t \right )} - 100.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos^{2}{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \left(\frac{d}{d t} a{\left (t \right )}\right)^{2} - 100.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} a{\left (t \right )} - 200.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \left(\frac{d}{d t} a{\left (t \right )}\right)^{2} - 25.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos^{2}{\left (\theta_{1}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{1}{\left (t \right )}\right)^{2} - 100.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} a{\left (t \right )} - 100.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \left(\frac{d}{d t} a{\left (t \right )}\right)^{2} + 100.0 \sin{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \sin{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} b{\left (t \right )} + 50.0 \sin{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \cos^{2}{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{1}{\left (t \right )}\right)^{2} + 100.0 \sin{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} a{\left (t \right )} + 50.0 \sin{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{1}{\left (t \right )}\right)^{2} + 100.0 \sin{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} a{\left (t \right )} + 200.0 \sin^{2}{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \frac{d}{d t} a{\left (t \right )} \frac{d}{d t} b{\left (t \right )} + 100.0 \sin{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \cos^{2}{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \frac{d^{2}}{d t^{2}} b{\left (t \right )} + 100.0 \sin{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} a{\left (t \right )} + 100.0 \sin{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} a{\left (t \right )} + 200.0 \sin{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \left(\frac{d}{d t} a{\left (t \right )}\right)^{2} + 1962.0 \sin{\left (\theta_{1}{\left (t \right )} \right )} \cos^{3}{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{2}{\left (t \right )} \right )} - 50.0 \sin{\left (\theta_{1}{\left (t \right )} \right )} \cos^{2}{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{1}{\left (t \right )}\right)^{2} - 200.0 \sin{\left (\theta_{2}{\left (t \right )} \right )} \cos^{4}{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} a{\left (t \right )} + 1962.0 \sin{\left (\theta_{2}{\left (t \right )} \right )} \cos^{3}{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} \right )} - 100.0 \sin{\left (\theta_{2}{\left (t \right )} \right )} \cos^{3}{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{2}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{1}{\left (t \right )}\right)^{2} + 400.0 \cos^{5}{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \frac{d}{d t} a{\left (t \right )} \frac{d}{d t} b{\left (t \right )} + 200.0 \cos^{4}{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{2}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} b{\left (t \right )} + 500.0 \cos^{3}{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \frac{d^{2}}{d t^{2}} \theta_{1}{\left (t \right )} + 50.0 \cos^{2}{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \frac{d^{2}}{d t^{2}} \theta_{1}{\left (t \right )} + 100.0 \cos^{2}{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \frac{d^{2}}{d t^{2}} a{\left (t \right )} + 50.0 \cos^{2}{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} \right )} \frac{d^{2}}{d t^{2}} \theta_{1}{\left (t \right )} + 100.0 \cos^{2}{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \frac{d^{2}}{d t^{2}} a{\left (t \right )} - 100.0 \cos{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \cos{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} b{\left (t \right )} - 200.0 \cos{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos^{2}{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \frac{d}{d t} a{\left (t \right )} \frac{d}{d t} b{\left (t \right )} - 100.0 \cos{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} b{\left (t \right )} - 200.0 \cos{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \cos{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{2}{\left (t \right )} \right )} \frac{d}{d t} a{\left (t \right )} \frac{d}{d t} b{\left (t \right )}\right) - 500.0 \left(\sin{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \frac{d^{2}}{d t^{2}} \theta_{1}{\left (t \right )} + \sin{\left (\theta_{1}{\left (t \right )} \right )} \frac{d^{2}}{d t^{2}} \theta_{1}{\left (t \right )} + \cos{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{1}{\left (t \right )}\right)^{2} + 2.0 \cos{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} \theta_{2}{\left (t \right )} + \cos{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{2}{\left (t \right )}\right)^{2} + \cos{\left (\theta_{1}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{1}{\left (t \right )}\right)^{2} + \cos{\left (\theta_{2}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{2}{\left (t \right )}\right)^{2} + 2.0 \frac{d^{2}}{d t^{2}} b{\left (t \right )}\right) \cos^{3}{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )}\right), \quad \lambda_{2} : \frac{0.005}{\left(\sin{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} + \sin{\left (\theta_{3}{\left (t \right )} \right )}\right)^{2} \cos^{3}{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )}} \left(\left(\sin{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} + \sin{\left (\theta_{3}{\left (t \right )} \right )}\right) \left(12.5 \left(\sin{\left (\theta_{1}{\left (t \right )} + 3.0 \theta_{3}{\left (t \right )} \right )} + \sin{\left (3.0 \theta_{1}{\left (t \right )} + 5.0 \theta_{3}{\left (t \right )} \right )}\right) \left(\frac{d}{d t} \theta_{1}{\left (t \right )}\right)^{2} + 50.0 \left(\sin{\left (3.0 \theta_{1}{\left (t \right )} + 3.0 \theta_{3}{\left (t \right )} \right )} + \sin{\left (5.0 \theta_{1}{\left (t \right )} + 5.0 \theta_{3}{\left (t \right )} \right )}\right) \left(\frac{d}{d t} a{\left (t \right )}\right)^{2} - 50.0 \left(\sin{\left (3.0 \theta_{1}{\left (t \right )} + 3.0 \theta_{3}{\left (t \right )} \right )} + \sin{\left (5.0 \theta_{1}{\left (t \right )} + 5.0 \theta_{3}{\left (t \right )} \right )}\right) \left(\frac{d}{d t} b{\left (t \right )}\right)^{2} - 50.0 \left(\cos{\left (3.0 \theta_{1}{\left (t \right )} + 3.0 \theta_{3}{\left (t \right )} \right )} - \cos{\left (5.0 \theta_{1}{\left (t \right )} + 5.0 \theta_{3}{\left (t \right )} \right )}\right) \frac{d}{d t} a{\left (t \right )} \frac{d}{d t} b{\left (t \right )} - 400.0 \sin^{2}{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos^{3}{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \frac{d}{d t} a{\left (t \right )} \frac{d}{d t} b{\left (t \right )} - 100.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \sin^{2}{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \left(\frac{d}{d t} b{\left (t \right )}\right)^{2} - 100.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \sin{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} b{\left (t \right )} - 100.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \sin{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} b{\left (t \right )} - 200.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \sin{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \frac{d}{d t} a{\left (t \right )} \frac{d}{d t} b{\left (t \right )} - 200.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \sin{\left (\theta_{3}{\left (t \right )} \right )} \cos^{3}{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} b{\left (t \right )} - 400.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos^{4}{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \left(\frac{d}{d t} a{\left (t \right )}\right)^{2} + 400.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos^{4}{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \left(\frac{d}{d t} b{\left (t \right )}\right)^{2} - 200.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos^{3}{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{3}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} a{\left (t \right )} - 25.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos^{2}{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{1}{\left (t \right )}\right)^{2} - 100.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \cos{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} a{\left (t \right )} - 50.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{1}{\left (t \right )}\right)^{2} - 100.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} a{\left (t \right )} - 100.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos^{2}{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \left(\frac{d}{d t} a{\left (t \right )}\right)^{2} - 100.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} a{\left (t \right )} - 200.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \left(\frac{d}{d t} a{\left (t \right )}\right)^{2} - 25.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos^{2}{\left (\theta_{1}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{1}{\left (t \right )}\right)^{2} - 100.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} a{\left (t \right )} - 100.0 \sin{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \left(\frac{d}{d t} a{\left (t \right )}\right)^{2} + 100.0 \sin{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \sin{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} b{\left (t \right )} + 50.0 \sin{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \cos^{2}{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{1}{\left (t \right )}\right)^{2} + 100.0 \sin{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} a{\left (t \right )} + 50.0 \sin{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{1}{\left (t \right )}\right)^{2} + 100.0 \sin{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} a{\left (t \right )} + 200.0 \sin^{2}{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \frac{d}{d t} a{\left (t \right )} \frac{d}{d t} b{\left (t \right )} + 100.0 \sin{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \cos^{2}{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \frac{d^{2}}{d t^{2}} b{\left (t \right )} + 100.0 \sin{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} a{\left (t \right )} + 100.0 \sin{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} a{\left (t \right )} + 200.0 \sin{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \left(\frac{d}{d t} a{\left (t \right )}\right)^{2} + 1962.0 \sin{\left (\theta_{1}{\left (t \right )} \right )} \cos^{3}{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{3}{\left (t \right )} \right )} - 50.0 \sin{\left (\theta_{1}{\left (t \right )} \right )} \cos^{2}{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{1}{\left (t \right )}\right)^{2} - 200.0 \sin{\left (\theta_{3}{\left (t \right )} \right )} \cos^{4}{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} a{\left (t \right )} + 1962.0 \sin{\left (\theta_{3}{\left (t \right )} \right )} \cos^{3}{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} \right )} - 100.0 \sin{\left (\theta_{3}{\left (t \right )} \right )} \cos^{3}{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{3}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{1}{\left (t \right )}\right)^{2} + 400.0 \cos^{5}{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \frac{d}{d t} a{\left (t \right )} \frac{d}{d t} b{\left (t \right )} + 200.0 \cos^{4}{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{3}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} b{\left (t \right )} + 500.0 \cos^{3}{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \frac{d^{2}}{d t^{2}} \theta_{1}{\left (t \right )} + 50.0 \cos^{2}{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \frac{d^{2}}{d t^{2}} \theta_{1}{\left (t \right )} + 100.0 \cos^{2}{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \frac{d^{2}}{d t^{2}} a{\left (t \right )} + 50.0 \cos^{2}{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} \right )} \frac{d^{2}}{d t^{2}} \theta_{1}{\left (t \right )} + 100.0 \cos^{2}{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \frac{d^{2}}{d t^{2}} a{\left (t \right )} - 100.0 \cos{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \cos{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} b{\left (t \right )} - 200.0 \cos{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos^{2}{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \frac{d}{d t} a{\left (t \right )} \frac{d}{d t} b{\left (t \right )} - 100.0 \cos{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \cos{\left (\theta_{1}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} b{\left (t \right )} - 200.0 \cos{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \cos{\left (2.0 \theta_{1}{\left (t \right )} + 2.0 \theta_{3}{\left (t \right )} \right )} \frac{d}{d t} a{\left (t \right )} \frac{d}{d t} b{\left (t \right )}\right) - 500.0 \left(\sin{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \frac{d^{2}}{d t^{2}} \theta_{1}{\left (t \right )} + \sin{\left (\theta_{1}{\left (t \right )} \right )} \frac{d^{2}}{d t^{2}} \theta_{1}{\left (t \right )} + \cos{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{1}{\left (t \right )}\right)^{2} + 2.0 \cos{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} \theta_{3}{\left (t \right )} + \cos{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{3}{\left (t \right )}\right)^{2} + \cos{\left (\theta_{1}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{1}{\left (t \right )}\right)^{2} + \cos{\left (\theta_{3}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{3}{\left (t \right )}\right)^{2} + 2.0 \frac{d^{2}}{d t^{2}} b{\left (t \right )}\right) \cos^{3}{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )}\right), \quad \frac{d^{2}}{d t^{2}} \theta_{2}{\left (t \right )} : - \frac{1}{\sin{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} + \sin{\left (\theta_{2}{\left (t \right )} \right )}} \left(\sin{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \frac{d^{2}}{d t^{2}} \theta_{1}{\left (t \right )} + \sin{\left (\theta_{1}{\left (t \right )} \right )} \frac{d^{2}}{d t^{2}} \theta_{1}{\left (t \right )} + \cos{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{1}{\left (t \right )}\right)^{2} + 2.0 \cos{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} \theta_{2}{\left (t \right )} + \cos{\left (\theta_{1}{\left (t \right )} + \theta_{2}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{2}{\left (t \right )}\right)^{2} + \cos{\left (\theta_{1}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{1}{\left (t \right )}\right)^{2} + \cos{\left (\theta_{2}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{2}{\left (t \right )}\right)^{2} + 2.0 \frac{d^{2}}{d t^{2}} b{\left (t \right )}\right), \quad \frac{d^{2}}{d t^{2}} \theta_{3}{\left (t \right )} : - \frac{1}{\sin{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} + \sin{\left (\theta_{3}{\left (t \right )} \right )}} \left(\sin{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \frac{d^{2}}{d t^{2}} \theta_{1}{\left (t \right )} + \sin{\left (\theta_{1}{\left (t \right )} \right )} \frac{d^{2}}{d t^{2}} \theta_{1}{\left (t \right )} + \cos{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{1}{\left (t \right )}\right)^{2} + 2.0 \cos{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \frac{d}{d t} \theta_{1}{\left (t \right )} \frac{d}{d t} \theta_{3}{\left (t \right )} + \cos{\left (\theta_{1}{\left (t \right )} + \theta_{3}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{3}{\left (t \right )}\right)^{2} + \cos{\left (\theta_{1}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{1}{\left (t \right )}\right)^{2} + \cos{\left (\theta_{3}{\left (t \right )} \right )} \left(\frac{d}{d t} \theta_{3}{\left (t \right )}\right)^{2} + 2.0 \frac{d^{2}}{d t^{2}} b{\left (t \right )}\right)\right \}$$
